scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/Job.java

```diff
@@ -358,6 +379,21 @@ public static Job parseJSONInfo(JSONObject jsonJobInfo) {
                        finishedTime);
     }
 
+    private static Map<String, String> extractGenericInformation(JSONValue jsonGenericInformation) {
+        Map<String, String> genericInformation = new HashMap<>();
+        if (jsonGenericInformation != null) {
+            JSONArray genericInformationArray = jsonGenericInformation.isArray();
+            if (genericInformationArray != null) {
+                for (int i = 0; i < genericInformationArray.size(); i++) {
```

Comment:
You could create the result Map with an initial capacity here and return an empty one in the other cases.